src/App.tsx

```diff
@@ -1,23 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
+const total = items.length;
+
 export const App: React.FC = () => {
+  const [itemsOnPage, setItemsOnPage] = useState(5);
+  const [selectedPage, setSelectedPage] = useState(1);
+
+  const preparedListOfItemsOnPage: string[][] = [];
+
+  for (let i = 0; i < total; i += itemsOnPage) {
+    preparedListOfItemsOnPage.push(items.slice(i, i + itemsOnPage));
+  }
+
+  const START = selectedPage * itemsOnPage - itemsOnPage + 1;
+  const END = (selectedPage - 1) * itemsOnPage
+    + preparedListOfItemsOnPage[selectedPage - 1].length;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${selectedPage} (items ${START} - ${END} of ${total})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            defaultValue="5"
+            onChange={event => {
+              setItemsOnPage(+event.target.value);
+
+              if (selectedPage !== 1) {
+                setSelectedPage(1);
+              }
+            }}
```

Comment:
it is more correct to move large functions into separate variables so as not to clutter jsx